gpMgmt/bin/gppylib/logfilter.py

```diff
@@ -543,14 +553,17 @@ def TimestampInBounds(iterable, begin, end):
 
     # Fetch first item from input stream.
     source = iter(iterable)
-    item = next(source)
+    try:
+        first_item = next(source)
+    except StopIteration:
+        return
 
     # If first item is a string, assume input consists of individual lines.
     # Yield lines which start with a timestamp within the given bounds, plus
     # any following lines which do not have timestamps.
-    if isinstance(item, str):
+    if isinstance(first_item, str):
         withinbounds = False
-        for item in source:
+        for item in chain(iter([first_item]), source):
```

Comment:
Am I right in understanding that we add `first_item` here because we extracted it from `source` above? Then why don't we do it below in a similar loop on `source`?

Reply:
Maybe rewrite both loops into one and check the type inside?

Reply:
Done